File: api/Ganymede.Api.Data/Spells/SpellRange.cs
using static Ganymede.Api.Data.Spells.Enums;

namespace Ganymede.Api.Data.Spells
{
    public class SpellRange
    {
        public int ID { get; set; }
        public int Amount { get; set; }
        public string Unit { get; set; }
        public string Shape { get; set; }
        public RangeType Type { get; set; }
    }
}

File: src/app/models/core/spells/spellRange.ts
export class SpellRange {
	public id: number;
	public amount: number;
	public unit: string;
	public self: boolean;
	public shape: string;
	public touch: boolean;

	public static getDefault(): SpellRange {
		const range = new SpellRange();
		range.id = -1;

		return range;
	}
}
